api/main.py

```diff
@@ -52,7 +52,27 @@ def get_user_info(user_id):
     user = api.user_repository.get_user_by_id(ObjectId(user_id))
     if user is None:
         return jsonify({"user": None, "error": {"code": 404, "message": "No such user found"}})
-    return jsonify({"user": user.get_user_attributes(), "error": None})
+    return jsonify({"user": {user.get_user_attributes(),['location']},"error": None})
+
+
+@api.route('/account/<user_id>/update', methods=['POST'])
+def update_user_data(user_id):
+    access = api.token_repository.authenticate_admin(get_request_token())
+    if not access:
+        return jsonify({"data": None, "error": {"code": 401, "message": "Authentication failed"}})
+    user = api.user_repository.get_user_by_id(ObjectId(user_id))
+    if user is None:
+        return jsonify({"success": False, "error": {"code": 404, "message": "No such user found"}})
+    data = request.get_json()
+    password = data['password']
+    name = user_id
```

Comment:
Why is the name the user_id?
You should get the other attributes from the data dictionary as well.